test/gold/cfg-test.clang.-g-grecord-gcc-switches-O1-O1-fno-discard-value-names.1-callsite.subset.callgraph.callgraph_edge.golden.csv

```diff
@@ -1,16 +1,13 @@
-nil	<cfg-test.c>:bar	[<<main-context>>, nil]	<cfg-test.c>:main:9
-nil	<cfg-test.c>:bar	nil	<cfg-test.c>:foo:1
```

Comment:
Triage: These changes are suspicious and should be reviewed before merging

Reply:
At `-O0`, these are very similar. At `-O1`, things start to change:

<details>
<summary>Comparison script</summary>

```
sudo \rm -rf 10 11
alias docker-here='docker run --rm -it --mount type=bind,src=$PWD,target=/work --workdir /work'
docker-here cclyzer-dev-llvm-11 clang-11 -O1 -O1 -fno-discard-value-names -S -emit-llvm test/c/cfg-test.c -o 11.ll
docker pull ghcr.io/galoisinc/cclyzerpp-dev:main
docker tag ghcr.io/galoisinc/cclyzerpp-dev:main cclyzer-dev
docker-here cclyzer-dev clang-10 -O1 -O1 -fno-discard-value-names -S -emit-llvm test/c/cfg-test.c -o 10.ll
docker-here cclyzer-dev-llvm-11 cmake -G Ninja -S . -B build -DLLVM_VERSION=11 -DCLANG_VERSION=11
docker-here cclyzer-dev-llvm-11 cmake --build build -j $(nproc) --target factgen-exe
docker-here cclyzer-dev-llvm-11 ./build/factgen-exe 11.ll --out-dir 11
sudo chown -R $(whoami):$(whoami) 11
for f in 11/*.gz; do gunzip "${f}"; done
for f in 11/*.csv; do sort "${f}" > "${f}.sorted"; \rm "${f}"; done
docker-here cclyzer-dev cmake -G Ninja -S . -B build-10 -DLLVM_VERSION=10 -DCLANG_VERSION=10
docker-here cclyzer-dev cmake --build build-10 -j $(nproc) --target factgen-exe
docker-here cclyzer-dev ./build/factgen-exe 10.ll --out-dir 10
sudo chown -R $(whoami):$(whoami) 10
for f in 10/*.gz; do gunzip "${f}"; done
for f in 10/*.csv; do sort "${f}" > "${f}.sorted"; \rm "${f}"; done
pushd 10
sed_recurse() { ag -g '.*' -0 | xargs -0 sed -E -i "$@"; }
sed_recurse 's|10\.ll|11\.ll|g'
popd
for f in 10/*.csv.sorted; do diff -u "${f}" 11/$(basename ${f}); done |& xsel -ib
```

</details>
<details>

<summary>10.ll</summary>

```
; ModuleID = 'test/c/cfg-test.c'
source_filename = "test/c/cfg-test.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@global = dso_local local_unnamed_addr global i32 0, align 4

; Function Attrs: norecurse nounwind readnone uwtable
define dso_local i32 @bar(i32 returned %x) local_unnamed_addr #0 {
entry:
  ret i32 %x
}

; Function Attrs: norecurse nounwind readonly uwtable
define dso_local i32 @foo(i32 %x) local_unnamed_addr #1 {
entry:
  %call = call i32 @bar(i32 %x)
  %add = shl i32 %call, 1
  %0 = load i32, i32* @global, align 4, !tbaa !2
  %add2 = add nsw i32 %add, %0
  ret i32 %add2
}

; Function Attrs: nofree norecurse nounwind uwtable writeonly
define dso_local void @baz() local_unnamed_addr #2 {
entry:
  store i32 1, i32* @global, align 4, !tbaa !2
  ret void
}

; Function Attrs: nofree norecurse nounwind uwtable
define dso_local i32 @main(i32 %argc, i8** nocapture readnone %argv) local_unnamed_addr #3 {
entry:
  %call = call i32 @foo(i32 0)
  call void @baz()
  %call1 = call i32 @foo(i32 1)
  %call2 = call i32 @bar(i32 %call1)
  %add = add nsw i32 %call1, %call
  %add3 = add nsw i32 %add, %call2
  ret i32 %add3
}

attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{!"clang version 10.0.0-4ubuntu1 "}
!2 = !{!3, !3, i64 0}
!3 = !{!"int", !4, i64 0}
!4 = !{!"omnipotent char", !5, i64 0}
!5 = !{!"Simple C/C++ TBAA"}
```

</details>

<summary>11.ll</summary>

```
; ModuleID = 'test/c/cfg-test.c'
source_filename = "test/c/cfg-test.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@global = dso_local local_unnamed_addr global i32 0, align 4

; Function Attrs: norecurse nounwind readnone uwtable
define dso_local i32 @bar(i32 returned %x) local_unnamed_addr #0 {
entry:
  ret i32 %x
}

; Function Attrs: norecurse nounwind readonly uwtable
define dso_local i32 @foo(i32 %x) local_unnamed_addr #1 {
entry:
  %add = shl nsw i32 %x, 1
  %0 = load i32, i32* @global, align 4, !tbaa !2
  %add2 = add nsw i32 %0, %add
  ret i32 %add2
}

; Function Attrs: nofree norecurse nounwind uwtable writeonly
define dso_local void @baz() local_unnamed_addr #2 {
entry:
  store i32 1, i32* @global, align 4, !tbaa !2
  ret void
}

; Function Attrs: nofree norecurse nounwind uwtable
define dso_local i32 @main(i32 %argc, i8** nocapture readnone %argv) local_unnamed_addr #3 {
entry:
  %call = call i32 @foo(i32 0)
  call void @baz()
  %call1 = call i32 @foo(i32 1)
  %reass.add = shl i32 %call1, 1
  %add3 = add i32 %reass.add, %call
  ret i32 %add3
}

attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{!"Ubuntu clang version 11.0.0-2~ubuntu20.04.1"}
!2 = !{!3, !3, i64 0}
!3 = !{!"int", !4, i64 0}
!4 = !{!"omnipotent char", !5, i64 0}
!5 = !{!"Simple C/C++ TBAA"}
```

</details>

<details>
<summary>LLVM diff</summary>

```diff
--- 10.ll	2022-10-25 15:12:32.090071789 -0400
+++ 11.ll	2022-10-25 15:12:31.413070210 -0400
@@ -14,10 +14,9 @@
 ; Function Attrs: norecurse nounwind readonly uwtable
 define dso_local i32 @foo(i32 %x) local_unnamed_addr #1 {
 entry:
-  %call = call i32 @bar(i32 %x)
-  %add = shl i32 %call, 1
+  %add = shl nsw i32 %x, 1
   %0 = load i32, i32* @global, align 4, !tbaa !2
-  %add2 = add nsw i32 %add, %0
+  %add2 = add nsw i32 %0, %add
   ret i32 %add2
 }
 
@@ -34,22 +33,21 @@
   %call = call i32 @foo(i32 0)
   call void @baz()
   %call1 = call i32 @foo(i32 1)
-  %call2 = call i32 @bar(i32 %call1)
-  %add = add nsw i32 %call1, %call
-  %add3 = add nsw i32 %add, %call2
+  %reass.add = shl i32 %call1, 1
+  %add3 = add i32 %reass.add, %call
   ret i32 %add3
 }
 
-attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
 
 !llvm.module.flags = !{!0}
 !llvm.ident = !{!1}
 
 !0 = !{i32 1, !"wchar_size", i32 4}
-!1 = !{!"clang version 10.0.0-4ubuntu1 "}
+!1 = !{!"Ubuntu clang version 11.0.0-2~ubuntu20.04.1"}
 !2 = !{!3, !3, i64 0}
 !3 = !{!"int", !4, i64 0}
 !4 = !{!"omnipotent char", !5, i64 0}
```

</details>

<details>
<summary>Relation diff</summary>

```diff
--- 10/add_instr.csv.sorted	2022-10-25 15:12:36.748082718 -0400
+++ 11/add_instr.csv.sorted	2022-10-25 15:12:33.822075842 -0400
@@ -1,3 +1,2 @@
-</work/11.ll>:foo:3
+</work/11.ll>:foo:2
 </work/11.ll>:main:4
-</work/11.ll>:main:5
--- 10/add_instr_first_operand.csv.sorted	2022-10-25 15:12:36.751082725 -0400
+++ 11/add_instr_first_operand.csv.sorted	2022-10-25 15:12:33.824075847 -0400
@@ -1,3 +1,2 @@
-</work/11.ll>:foo:3	</work/11.ll>:foo:%add
-</work/11.ll>:main:4	</work/11.ll>:main:%call1
-</work/11.ll>:main:5	</work/11.ll>:main:%add
+</work/11.ll>:foo:2	</work/11.ll>:foo:%0
+</work/11.ll>:main:4	</work/11.ll>:main:%reass.add
--- 10/add_instr_second_operand.csv.sorted	2022-10-25 15:12:36.746082714 -0400
+++ 11/add_instr_second_operand.csv.sorted	2022-10-25 15:12:33.826075851 -0400
@@ -1,3 +1,2 @@
-</work/11.ll>:foo:3	</work/11.ll>:foo:%0
+</work/11.ll>:foo:2	</work/11.ll>:foo:%add
 </work/11.ll>:main:4	</work/11.ll>:main:%call
-</work/11.ll>:main:5	</work/11.ll>:main:%call2
--- 10/call_instr_arg.csv.sorted	2022-10-25 15:12:36.753082730 -0400
+++ 11/call_instr_arg.csv.sorted	2022-10-25 15:12:33.895076013 -0400
@@ -1,4 +1,2 @@
-</work/11.ll>:foo:0	0	</work/11.ll>:foo:%x
 </work/11.ll>:main:0	0	</work/11.ll>:main:0:1:0
 </work/11.ll>:main:2	0	</work/11.ll>:main:2:1:1
-</work/11.ll>:main:3	0	</work/11.ll>:main:%call1
--- 10/call_instr.csv.sorted	2022-10-25 15:12:36.751082725 -0400
+++ 11/call_instr.csv.sorted	2022-10-25 15:12:33.899076022 -0400
@@ -1,5 +1,3 @@
-</work/11.ll>:foo:0
 </work/11.ll>:main:0
 </work/11.ll>:main:1
 </work/11.ll>:main:2
-</work/11.ll>:main:3
--- 10/call_instr_func_operand.csv.sorted	2022-10-25 15:12:36.751082725 -0400
+++ 11/call_instr_func_operand.csv.sorted	2022-10-25 15:12:33.902076029 -0400
@@ -1,5 +1,3 @@
-</work/11.ll>:foo:0	</work/11.ll>:foo:0:0:@bar
 </work/11.ll>:main:0	</work/11.ll>:main:0:0:@foo
 </work/11.ll>:main:1	</work/11.ll>:main:1:0:@baz
 </work/11.ll>:main:2	</work/11.ll>:main:2:0:@foo
-</work/11.ll>:main:3	</work/11.ll>:main:3:0:@bar
--- 10/constant.csv.sorted	2022-10-25 15:12:36.747082716 -0400
+++ 11/constant.csv.sorted	2022-10-25 15:12:33.927076088 -0400
@@ -1,12 +1,11 @@
 </work/11.ll>:0:0
 </work/11.ll>:baz:0:0:1
 </work/11.ll>:baz:0:1:@global
-</work/11.ll>:foo:0:0:@bar
-</work/11.ll>:foo:1:0:1
-</work/11.ll>:foo:2:0:@global
+</work/11.ll>:foo:0:0:1
+</work/11.ll>:foo:1:0:@global
 </work/11.ll>:main:0:0:@foo
 </work/11.ll>:main:0:1:0
 </work/11.ll>:main:1:0:@baz
 </work/11.ll>:main:2:0:@foo
 </work/11.ll>:main:2:1:1
-</work/11.ll>:main:3:0:@bar
+</work/11.ll>:main:3:0:1
--- 10/constant_hashes_to.csv.sorted	2022-10-25 15:12:36.748082718 -0400
+++ 11/constant_hashes_to.csv.sorted	2022-10-25 15:12:33.930076095 -0400
@@ -1,12 +1,11 @@
 </work/11.ll>:0:0	4490122437319657922
 </work/11.ll>:baz:0:0:1	4632970265397325641
 </work/11.ll>:baz:0:1:@global	6005532478192098633
-</work/11.ll>:foo:0:0:@bar	2938245187859127749
-</work/11.ll>:foo:1:0:1	4632970265397325641
-</work/11.ll>:foo:2:0:@global	6005532478192098633
+</work/11.ll>:foo:0:0:1	4632970265397325641
+</work/11.ll>:foo:1:0:@global	6005532478192098633
 </work/11.ll>:main:0:0:@foo	3067270429617558524
 </work/11.ll>:main:0:1:0	4490122437319657922
 </work/11.ll>:main:1:0:@baz	15884898656262208119
 </work/11.ll>:main:2:0:@foo	3067270429617558524
 </work/11.ll>:main:2:1:1	4632970265397325641
-</work/11.ll>:main:3:0:@bar	2938245187859127749
+</work/11.ll>:main:3:0:1	4632970265397325641
--- 10/constant_has_type.csv.sorted	2022-10-25 15:12:36.753082730 -0400
+++ 11/constant_has_type.csv.sorted	2022-10-25 15:12:33.932076100 -0400
@@ -1,12 +1,11 @@
 </work/11.ll>:0:0	i32
 </work/11.ll>:baz:0:0:1	i32
 </work/11.ll>:baz:0:1:@global	i32*
-</work/11.ll>:foo:0:0:@bar	i32 (i32)*
-</work/11.ll>:foo:1:0:1	i32
-</work/11.ll>:foo:2:0:@global	i32*
+</work/11.ll>:foo:0:0:1	i32
+</work/11.ll>:foo:1:0:@global	i32*
 </work/11.ll>:main:0:0:@foo	i32 (i32)*
 </work/11.ll>:main:0:1:0	i32
 </work/11.ll>:main:1:0:@baz	void ()*
 </work/11.ll>:main:2:0:@foo	i32 (i32)*
 </work/11.ll>:main:2:1:1	i32
-</work/11.ll>:main:3:0:@bar	i32 (i32)*
+</work/11.ll>:main:3:0:1	i32
--- 10/constant_has_value.csv.sorted	2022-10-25 15:12:36.750082723 -0400
+++ 11/constant_has_value.csv.sorted	2022-10-25 15:12:33.934076104 -0400
@@ -1,12 +1,11 @@
 </work/11.ll>:0:0	0
 </work/11.ll>:baz:0:0:1	1
 </work/11.ll>:baz:0:1:@global	@global
-</work/11.ll>:foo:0:0:@bar	@bar
-</work/11.ll>:foo:1:0:1	1
-</work/11.ll>:foo:2:0:@global	@global
+</work/11.ll>:foo:0:0:1	1
+</work/11.ll>:foo:1:0:@global	@global
 </work/11.ll>:main:0:0:@foo	@foo
 </work/11.ll>:main:0:1:0	0
 </work/11.ll>:main:1:0:@baz	@baz
 </work/11.ll>:main:2:0:@foo	@foo
 </work/11.ll>:main:2:1:1	1
-</work/11.ll>:main:3:0:@bar	@bar
+</work/11.ll>:main:3:0:1	1
--- 10/constant_in_func_name.csv.sorted	2022-10-25 15:12:36.753082730 -0400
+++ 11/constant_in_func_name.csv.sorted	2022-10-25 15:12:33.936076109 -0400
@@ -1,11 +1,10 @@
 </work/11.ll>:baz:0:0:1	@baz
 </work/11.ll>:baz:0:1:@global	@baz
-</work/11.ll>:foo:0:0:@bar	@foo
-</work/11.ll>:foo:1:0:1	@foo
-</work/11.ll>:foo:2:0:@global	@foo
+</work/11.ll>:foo:0:0:1	@foo
+</work/11.ll>:foo:1:0:@global	@foo
 </work/11.ll>:main:0:0:@foo	@main
 </work/11.ll>:main:0:1:0	@main
 </work/11.ll>:main:1:0:@baz	@main
 </work/11.ll>:main:2:0:@foo	@main
 </work/11.ll>:main:2:1:1	@main
-</work/11.ll>:main:3:0:@bar	@main
+</work/11.ll>:main:3:0:1	@main
--- 10/constant_to_int.csv.sorted	2022-10-25 15:12:36.750082723 -0400
+++ 11/constant_to_int.csv.sorted	2022-10-25 15:12:33.943076125 -0400
@@ -1,5 +1,6 @@
 </work/11.ll>:0:0	0
 </work/11.ll>:baz:0:0:1	1
-</work/11.ll>:foo:1:0:1	1
+</work/11.ll>:foo:0:0:1	1
 </work/11.ll>:main:0:1:0	0
 </work/11.ll>:main:2:1:1	1
+</work/11.ll>:main:3:0:1	1
--- 10/func_constant.csv.sorted	2022-10-25 15:12:36.751082725 -0400
+++ 11/func_constant.csv.sorted	2022-10-25 15:12:34.027076322 -0400
@@ -1,5 +1,3 @@
-</work/11.ll>:foo:0:0:@bar
 </work/11.ll>:main:0:0:@foo
 </work/11.ll>:main:1:0:@baz
 </work/11.ll>:main:2:0:@foo
-</work/11.ll>:main:3:0:@bar
--- 10/func_constant_fn_name.csv.sorted	2022-10-25 15:12:36.749082721 -0400
+++ 11/func_constant_fn_name.csv.sorted	2022-10-25 15:12:34.029076326 -0400
@@ -1,5 +1,3 @@
-</work/11.ll>:foo:0:0:@bar	@bar
 </work/11.ll>:main:0:0:@foo	@foo
 </work/11.ll>:main:1:0:@baz	@baz
 </work/11.ll>:main:2:0:@foo	@foo
-</work/11.ll>:main:3:0:@bar	@bar
--- 10/func_func_attr.csv.sorted	2022-10-25 15:12:36.748082718 -0400
+++ 11/func_func_attr.csv.sorted	2022-10-25 15:12:34.034076339 -0400
@@ -8,7 +8,7 @@
 </work/11.ll>:bar	no-nans-fp-math=false
 </work/11.ll>:bar	norecurse
 </work/11.ll>:bar	no-signed-zeros-fp-math=false
-</work/11.ll>:bar	no-trapping-math=false
+</work/11.ll>:bar	no-trapping-math=true
 </work/11.ll>:bar	nounwind
 </work/11.ll>:bar	readnone
 </work/11.ll>:bar	stack-protector-buffer-size=8
@@ -28,7 +28,7 @@
 </work/11.ll>:baz	no-nans-fp-math=false
 </work/11.ll>:baz	norecurse
 </work/11.ll>:baz	no-signed-zeros-fp-math=false
-</work/11.ll>:baz	no-trapping-math=false
+</work/11.ll>:baz	no-trapping-math=true
 </work/11.ll>:baz	nounwind
 </work/11.ll>:baz	stack-protector-buffer-size=8
 </work/11.ll>:baz	target-cpu=x86-64
@@ -47,7 +47,7 @@
 </work/11.ll>:foo	no-nans-fp-math=false
 </work/11.ll>:foo	norecurse
 </work/11.ll>:foo	no-signed-zeros-fp-math=false
-</work/11.ll>:foo	no-trapping-math=false
+</work/11.ll>:foo	no-trapping-math=true
 </work/11.ll>:foo	nounwind
 </work/11.ll>:foo	readonly
 </work/11.ll>:foo	stack-protector-buffer-size=8
@@ -67,7 +67,7 @@
 </work/11.ll>:main	no-nans-fp-math=false
 </work/11.ll>:main	norecurse
 </work/11.ll>:main	no-signed-zeros-fp-math=false
-</work/11.ll>:main	no-trapping-math=false
+</work/11.ll>:main	no-trapping-math=true
 </work/11.ll>:main	nounwind
 </work/11.ll>:main	stack-protector-buffer-size=8
 </work/11.ll>:main	target-cpu=x86-64
--- 10/global_var_constant.csv.sorted	2022-10-25 15:12:36.750082723 -0400
+++ 11/global_var_constant.csv.sorted	2022-10-25 15:12:34.084076456 -0400
@@ -1,2 +1,2 @@
 </work/11.ll>:baz:0:1:@global
-</work/11.ll>:foo:2:0:@global
+</work/11.ll>:foo:1:0:@global
--- 10/global_var_constant_name.csv.sorted	2022-10-25 15:12:36.751082725 -0400
+++ 11/global_var_constant_name.csv.sorted	2022-10-25 15:12:34.086076460 -0400
@@ -1,2 +1,2 @@
 </work/11.ll>:baz:0:1:@global	@global
-</work/11.ll>:foo:2:0:@global	@global
+</work/11.ll>:foo:1:0:@global	@global
--- 10/instr_assigns_to.csv.sorted	2022-10-25 15:12:36.746082714 -0400
+++ 11/instr_assigns_to.csv.sorted	2022-10-25 15:12:34.141076590 -0400
@@ -1,9 +1,7 @@
-</work/11.ll>:foo:0	</work/11.ll>:foo:%call
-</work/11.ll>:foo:1	</work/11.ll>:foo:%add
-</work/11.ll>:foo:2	</work/11.ll>:foo:%0
-</work/11.ll>:foo:3	</work/11.ll>:foo:%add2
+</work/11.ll>:foo:0	</work/11.ll>:foo:%add
+</work/11.ll>:foo:1	</work/11.ll>:foo:%0
+</work/11.ll>:foo:2	</work/11.ll>:foo:%add2
 </work/11.ll>:main:0	</work/11.ll>:main:%call
 </work/11.ll>:main:2	</work/11.ll>:main:%call1
-</work/11.ll>:main:3	</work/11.ll>:main:%call2
-</work/11.ll>:main:4	</work/11.ll>:main:%add
-</work/11.ll>:main:5	</work/11.ll>:main:%add3
+</work/11.ll>:main:3	</work/11.ll>:main:%reass.add
+</work/11.ll>:main:4	</work/11.ll>:main:%add3
--- 10/instr_bb_entry.csv.sorted	2022-10-25 15:12:36.753082730 -0400
+++ 11/instr_bb_entry.csv.sorted	2022-10-25 15:12:34.143076594 -0400
@@ -5,11 +5,9 @@
 </work/11.ll>:foo:1	</work/11.ll>:foo:[basicblock]%entry
 </work/11.ll>:foo:2	</work/11.ll>:foo:[basicblock]%entry
 </work/11.ll>:foo:3	</work/11.ll>:foo:[basicblock]%entry
-</work/11.ll>:foo:4	</work/11.ll>:foo:[basicblock]%entry
 </work/11.ll>:main:0	</work/11.ll>:main:[basicblock]%entry
 </work/11.ll>:main:1	</work/11.ll>:main:[basicblock]%entry
 </work/11.ll>:main:2	</work/11.ll>:main:[basicblock]%entry
 </work/11.ll>:main:3	</work/11.ll>:main:[basicblock]%entry
 </work/11.ll>:main:4	</work/11.ll>:main:[basicblock]%entry
 </work/11.ll>:main:5	</work/11.ll>:main:[basicblock]%entry
-</work/11.ll>:main:6	</work/11.ll>:main:[basicblock]%entry
--- 10/instr_flag.csv.sorted	2022-10-25 15:12:36.751082725 -0400
+++ 11/instr_flag.csv.sorted	2022-10-25 15:12:34.144076597 -0400
@@ -1,3 +1,2 @@
-</work/11.ll>:foo:3	nsw
-</work/11.ll>:main:4	nsw
-</work/11.ll>:main:5	nsw
+</work/11.ll>:foo:0	nsw
+</work/11.ll>:foo:2	nsw
--- 10/instr_func.csv.sorted	2022-10-25 15:12:36.751082725 -0400
+++ 11/instr_func.csv.sorted	2022-10-25 15:12:34.146076601 -0400
@@ -5,11 +5,9 @@
 </work/11.ll>:foo:1	</work/11.ll>:foo
 </work/11.ll>:foo:2	</work/11.ll>:foo
 </work/11.ll>:foo:3	</work/11.ll>:foo
-</work/11.ll>:foo:4	</work/11.ll>:foo
 </work/11.ll>:main:0	</work/11.ll>:main
 </work/11.ll>:main:1	</work/11.ll>:main
 </work/11.ll>:main:2	</work/11.ll>:main
 </work/11.ll>:main:3	</work/11.ll>:main
 </work/11.ll>:main:4	</work/11.ll>:main
 </work/11.ll>:main:5	</work/11.ll>:main
-</work/11.ll>:main:6	</work/11.ll>:main
--- 10/instr_successor.csv.sorted	2022-10-25 15:12:36.746082714 -0400
+++ 11/instr_successor.csv.sorted	2022-10-25 15:12:34.150076611 -0400
@@ -2,10 +2,8 @@
 </work/11.ll>:foo:0	</work/11.ll>:foo:1
 </work/11.ll>:foo:1	</work/11.ll>:foo:2
 </work/11.ll>:foo:2	</work/11.ll>:foo:3
-</work/11.ll>:foo:3	</work/11.ll>:foo:4
 </work/11.ll>:main:0	</work/11.ll>:main:1
 </work/11.ll>:main:1	</work/11.ll>:main:2
 </work/11.ll>:main:2	</work/11.ll>:main:3
 </work/11.ll>:main:3	</work/11.ll>:main:4
 </work/11.ll>:main:4	</work/11.ll>:main:5
-</work/11.ll>:main:5	</work/11.ll>:main:6
--- 10/integer_constant.csv.sorted	2022-10-25 15:12:36.750082723 -0400
+++ 11/integer_constant.csv.sorted	2022-10-25 15:12:34.151076613 -0400
@@ -1,5 +1,6 @@
 </work/11.ll>:0:0
 </work/11.ll>:baz:0:0:1
-</work/11.ll>:foo:1:0:1
+</work/11.ll>:foo:0:0:1
 </work/11.ll>:main:0:1:0
 </work/11.ll>:main:2:1:1
+</work/11.ll>:main:3:0:1
--- 10/load_instr_address.csv.sorted	2022-10-25 15:12:36.749082721 -0400
+++ 11/load_instr_address.csv.sorted	2022-10-25 15:12:34.190076705 -0400
@@ -1 +1 @@
-</work/11.ll>:foo:2	</work/11.ll>:foo:2:0:@global
+</work/11.ll>:foo:1	</work/11.ll>:foo:1:0:@global
--- 10/load_instr_alignment.csv.sorted	2022-10-25 15:12:36.747082716 -0400
+++ 11/load_instr_alignment.csv.sorted	2022-10-25 15:12:34.192076709 -0400
@@ -1 +1 @@
-</work/11.ll>:foo:2	4
+</work/11.ll>:foo:1	4
--- 10/load_instr.csv.sorted	2022-10-25 15:12:36.751082725 -0400
+++ 11/load_instr.csv.sorted	2022-10-25 15:12:34.194076714 -0400
@@ -1 +1 @@
-</work/11.ll>:foo:2
+</work/11.ll>:foo:1
--- 10/ret_instr.csv.sorted	2022-10-25 15:12:36.753082730 -0400
+++ 11/ret_instr.csv.sorted	2022-10-25 15:12:34.248076841 -0400
@@ -1,4 +1,4 @@
 </work/11.ll>:bar:0
 </work/11.ll>:baz:1
-</work/11.ll>:foo:4
-</work/11.ll>:main:6
+</work/11.ll>:foo:3
+</work/11.ll>:main:5
--- 10/ret_instr_operand.csv.sorted	2022-10-25 15:12:36.754082733 -0400
+++ 11/ret_instr_operand.csv.sorted	2022-10-25 15:12:34.250076845 -0400
@@ -1,3 +1,3 @@
 </work/11.ll>:bar:0	</work/11.ll>:bar:%x
-</work/11.ll>:foo:4	</work/11.ll>:foo:%add2
-</work/11.ll>:main:6	</work/11.ll>:main:%add3
+</work/11.ll>:foo:3	</work/11.ll>:foo:%add2
+</work/11.ll>:main:5	</work/11.ll>:main:%add3
--- 10/shl_instr.csv.sorted	2022-10-25 15:12:36.748082718 -0400
+++ 11/shl_instr.csv.sorted	2022-10-25 15:12:34.271076895 -0400
@@ -1 +1,2 @@
-</work/11.ll>:foo:1
+</work/11.ll>:foo:0
+</work/11.ll>:main:3
--- 10/shl_instr_first_operand.csv.sorted	2022-10-25 15:12:36.752082728 -0400
+++ 11/shl_instr_first_operand.csv.sorted	2022-10-25 15:12:34.273076899 -0400
@@ -1 +1,2 @@
-</work/11.ll>:foo:1	</work/11.ll>:foo:%call
+</work/11.ll>:foo:0	</work/11.ll>:foo:%x
+</work/11.ll>:main:3	</work/11.ll>:main:%call1
--- 10/shl_instr_second_operand.csv.sorted	2022-10-25 15:12:36.754082733 -0400
+++ 11/shl_instr_second_operand.csv.sorted	2022-10-25 15:12:34.275076903 -0400
@@ -1 +1,2 @@
-</work/11.ll>:foo:1	</work/11.ll>:foo:1:0:1
+</work/11.ll>:foo:0	</work/11.ll>:foo:0:0:1
+</work/11.ll>:main:3	</work/11.ll>:main:3:0:1
--- 10/target_dependent_attr.csv.sorted	2022-10-25 15:12:36.750082723 -0400
+++ 11/target_dependent_attr.csv.sorted	2022-10-25 15:12:34.361077106 -0400
@@ -34,10 +34,10 @@
 no-signed-zeros-fp-math=false
 no-signed-zeros-fp-math=false
 no-signed-zeros-fp-math=false
-no-trapping-math=false
-no-trapping-math=false
-no-trapping-math=false
-no-trapping-math=false
+no-trapping-math=true
+no-trapping-math=true
+no-trapping-math=true
+no-trapping-math=true
 stack-protector-buffer-size=8
 stack-protector-buffer-size=8
 stack-protector-buffer-size=8
--- 10/variable.csv.sorted	2022-10-25 15:12:36.750082723 -0400
+++ 11/variable.csv.sorted	2022-10-25 15:12:34.398077193 -0400
@@ -5,13 +5,11 @@
 </work/11.ll>:foo:%add
 </work/11.ll>:foo:%add2
 </work/11.ll>:foo:[basicblock]%entry
-</work/11.ll>:foo:%call
 </work/11.ll>:foo:%x
-</work/11.ll>:main:%add
 </work/11.ll>:main:%add3
 </work/11.ll>:main:%argc
 </work/11.ll>:main:%argv
 </work/11.ll>:main:[basicblock]%entry
 </work/11.ll>:main:%call
 </work/11.ll>:main:%call1
-</work/11.ll>:main:%call2
+</work/11.ll>:main:%reass.add
--- 10/variable_has_name.csv.sorted	2022-10-25 15:12:36.753082730 -0400
+++ 11/variable_has_name.csv.sorted	2022-10-25 15:12:34.403077204 -0400
@@ -5,13 +5,11 @@
 </work/11.ll>:foo:%add2	%add2
 </work/11.ll>:foo:%add	%add
 </work/11.ll>:foo:[basicblock]%entry	%entry
-</work/11.ll>:foo:%call	%call
 </work/11.ll>:foo:%x	%x
 </work/11.ll>:main:%add3	%add3
-</work/11.ll>:main:%add	%add
 </work/11.ll>:main:%argc	%argc
 </work/11.ll>:main:%argv	%argv
 </work/11.ll>:main:[basicblock]%entry	%entry
 </work/11.ll>:main:%call1	%call1
-</work/11.ll>:main:%call2	%call2
 </work/11.ll>:main:%call	%call
+</work/11.ll>:main:%reass.add	%reass.add
--- 10/variable_has_type.csv.sorted	2022-10-25 15:12:36.747082716 -0400
+++ 11/variable_has_type.csv.sorted	2022-10-25 15:12:34.405077209 -0400
@@ -5,13 +5,11 @@
 </work/11.ll>:foo:%add2	i32
 </work/11.ll>:foo:%add	i32
 </work/11.ll>:foo:[basicblock]%entry	label
-</work/11.ll>:foo:%call	i32
 </work/11.ll>:foo:%x	i32
 </work/11.ll>:main:%add3	i32
-</work/11.ll>:main:%add	i32
 </work/11.ll>:main:%argc	i32
 </work/11.ll>:main:%argv	i8**
 </work/11.ll>:main:[basicblock]%entry	label
 </work/11.ll>:main:%call1	i32
-</work/11.ll>:main:%call2	i32
 </work/11.ll>:main:%call	i32
+</work/11.ll>:main:%reass.add	i32
--- 10/variable_in_func_name.csv.sorted	2022-10-25 15:12:36.749082721 -0400
+++ 11/variable_in_func_name.csv.sorted	2022-10-25 15:12:34.407077214 -0400
@@ -5,13 +5,11 @@
 </work/11.ll>:foo:%add2	@foo
 </work/11.ll>:foo:%add	@foo
 </work/11.ll>:foo:[basicblock]%entry	foo
-</work/11.ll>:foo:%call	@foo
 </work/11.ll>:foo:%x	@foo
 </work/11.ll>:main:%add3	@main
-</work/11.ll>:main:%add	@main
 </work/11.ll>:main:%argc	@main
 </work/11.ll>:main:%argv	@main
 </work/11.ll>:main:[basicblock]%entry	main
 </work/11.ll>:main:%call1	@main
-</work/11.ll>:main:%call2	@main
 </work/11.ll>:main:%call	@main
+</work/11.ll>:main:%reass.add	@main
```

</details>

Reply:
Based on the LLVM diff, these changes look valid. It looks like LLVM 11 is choosing to inline `bar` at `-O1` which is completely valid:
```c
int bar(int x) { return x; }
```